File: app/src/main/java/com/flatstack/android/MainActivity.java
package com.flatstack.android;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.Toolbar;

import com.flatstack.android.fragments.MainFragmentBuilder;
import com.flatstack.android.rx.RxActivity;

import butterknife.Bind;

public class MainActivity extends RxActivity {

    @Bind(R.id.toolbar) Toolbar toolbar;

    @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setSupportActionBar(toolbar);
        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                .replace(R.id.content, new MainFragmentBuilder(R.id.content).build())
                .commit();
        }

    }
}

File: app/src/main/java/com/flatstack/android/mainscreen/MainActivity.kt
package com.flatstack.android.mainscreen

import android.os.Bundle
import com.flatstack.android.R
import com.flatstack.android.utils.Intents
import com.flatstack.android.utils.ui.BaseActivity
import com.flatstack.android.utils.ui.ScreenConfig

class MainActivity : BaseActivity() {
    override val screenConfig: ScreenConfig
        get() = ScreenConfig(R.layout.activity_main, titleRes = R.string.app_name, enableBackButton = true)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        when {
            Intents.isActivityExpandedFromLauncherIcon(this) -> finish()
            savedInstanceState == null -> showMainFragment()
        }
    }

    private fun showMainFragment() {
        supportFragmentManager.beginTransaction()
            .replace(R.id.content, MainFragment())
            .commit()
    }
}
